pkg/helpers/helpers.go

```diff
@@ -167,16 +167,12 @@ func IsBpfApplicationConditionFailure(conditions *[]metav1.Condition) bool {
 		log.Info("more than one condition found", "numConditions", numConditions)
 	}
 
-	if (*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppStateCondPending) ||
-		(*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppStateCondError) ||
-		(*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppStateCondProgramListChangedError) {
-		return true
-	}
-
-	return false
+	return (*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppStateCondError) ||
```

Comment:
r we sure condition[0] will always have error in this case can condition[0] be success but [1] is the failing one ? maybe we can just loop and if any of the condition match what u have we return it ?

Reply:
The way we use conditions is wonky and we should fix it, but as it is, there should only ever be 1 condition and the actual condition should always be at index 0.  See https://github.com/bpfman/bpfman-operator/blob/main/controllers/bpfman-agent/common.go#L160 for the details.  It's an internal bug if `numConditions > 1` here, so we log it, but it shouldn't ever happen with the current code.